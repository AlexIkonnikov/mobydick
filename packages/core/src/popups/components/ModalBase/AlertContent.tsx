import React, {FC} from 'react';
import {ViewStyle} from 'react-native';

import SimpleIcon, {
  SimpleIconName,
} from '../../../styles/icons/font/SimpleIcon';
import useStyles from '../../../styles/theme/hooks/useStyles';
import useTheme from '../../../styles/theme/hooks/useTheme';
import View from '../../../basic/components/View/View';
import rem from '../../../styles/spaces/rem';

import stylesCreate from './stylesCreate';

interface IProps {
  name?: SimpleIconName;
  color?: string;
  size?: number;
  style?: ViewStyle | ViewStyle[];
}

const AlertContent: FC<IProps> = props => {
  const [styles] = useStyles(stylesCreate);
  const {colors} = useTheme();
  const {name, color, size, style} = props;

  return (
    <View style={[styles.alertView, style]}>
      <SimpleIcon
        name={name || 'icon-check'}
        size={size || rem(36)}
        color={color || colors.IconBase}
      />
    </View>
  );
};

export default AlertContent;
