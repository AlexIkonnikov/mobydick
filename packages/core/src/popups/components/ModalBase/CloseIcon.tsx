import React, {FC} from 'react';

import {IPopupCloseIcon} from '../PopupBase';
import useStyles from '../../../styles/theme/hooks/useStyles';
import TouchableOpacity from '../../../basic/components/TouchableOpacity/TouchableOpacity';
import SimpleIcon from '../../../styles/icons/font/SimpleIcon';

import stylesCreate from './stylesCreate';

const CloseIcon: FC<IPopupCloseIcon> = props => {
  const [styles] = useStyles(stylesCreate);
  const {onPress} = props;

  return (
    <TouchableOpacity style={styles.closeButton} onPress={onPress}>
      <SimpleIcon name={'icon-cancel'} />
    </TouchableOpacity>
  );
};

export default CloseIcon;
