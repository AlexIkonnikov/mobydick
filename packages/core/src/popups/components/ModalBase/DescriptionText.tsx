import React, {FC} from 'react';

import {IDescriptionTextPopup} from '../PopupBase';
import useStyles from '../../../styles/theme/hooks/useStyles';
import {Typography} from '../../../typography';

import stylesCreate from './stylesCreate';

/**
 * @deprecated use TextContent.tsx
 */
const DescriptionText: FC<IDescriptionTextPopup> = props => {
  const {descriptionText, descriptionStyles, descriptionFont} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <Typography
      style={[styles.text, descriptionStyles]}
      font={descriptionFont ? descriptionFont : 'Regular-Tertiary-XS'}>
      {descriptionText}
    </Typography>
  );
};

export default DescriptionText;
