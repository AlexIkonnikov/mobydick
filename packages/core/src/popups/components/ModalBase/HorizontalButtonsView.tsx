import {FC} from 'react';

import {IHorizontalButtonsView} from '../../types';
import useStyles from '../../../styles/hooks/useStyles';
import View from '../../../basic/components/View/View';
import Button from '../../../cta/components/Button/Button';
import {IButtonSize} from '../../../cta/components/Button/types';

import stylesCreate from './stylesCreate';

const HorizontalButtonsView: FC<IHorizontalButtonsView> = props => {
  const [styles] = useStyles(stylesCreate);
  const {
    typeLeft,
    onPressLeft,
    textLeft,
    typeRight,
    textRight,
    onPressRight,
    disabledRight,
    disabledLeft,
  } = props;

  return (
    <View style={styles.horizontalButtonsView}>
      <Button
        size={IButtonSize.fixed}
        style={styles.horizontalLeftButton}
        type={typeLeft}
        onPress={onPressLeft}
        text={textLeft}
        disabled={disabledLeft}
      />
      <Button
        size={IButtonSize.fixed}
        style={styles.horizontalRightButton}
        type={typeRight}
        onPress={onPressRight}
        text={textRight}
        disabled={disabledRight}
      />
    </View>
  );
};

export default HorizontalButtonsView;
