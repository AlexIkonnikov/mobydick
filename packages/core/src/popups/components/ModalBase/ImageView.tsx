import React, {FC, ReactElement} from 'react';
import {ViewStyle} from 'react-native';

import useStyles from '../../../styles/theme/hooks/useStyles';
import View from '../../../basic/components/View/View';

import stylesCreate from './stylesCreate';

interface IProps {
  image: ReactElement;
  imageStyles?: ViewStyle | ViewStyle[];
}

const ImageView: FC<IProps> = props => {
  const [styles] = useStyles(stylesCreate);

  return (
    <View style={[styles.imageView, props.imageStyles]}>{props.image}</View>
  );
};

export default ImageView;
