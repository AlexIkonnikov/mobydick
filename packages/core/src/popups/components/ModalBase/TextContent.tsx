import React, {FC} from 'react';

import {ITextContentProps} from '../PopupBase';
import useStyles from '../../../styles/theme/hooks/useStyles';
import View from '../../../basic/components/View/View';
import {Typography} from '../../../typography/components/Typography/Typography';

import stylesCreate from './stylesCreate';

const TextContent: FC<ITextContentProps> = props => {
  const {
    title,
    titleStyles,
    titleFont,
    descriptionText,
    descriptionStyles,
    descriptionFont,
  } = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <View style={styles.textContent}>
      {Boolean(title) && (
        <Typography
          style={[styles.title, titleStyles]}
          font={titleFont ? titleFont : 'SemiBold-Primary-L'}>
          {title}
        </Typography>
      )}
      {Boolean(descriptionText) && (
        <Typography
          style={[styles.description, descriptionStyles]}
          font={descriptionFont ? descriptionFont : 'Regular-Tertiary-XS'}>
          {descriptionText}
        </Typography>
      )}
    </View>
  );
};
export default TextContent;
