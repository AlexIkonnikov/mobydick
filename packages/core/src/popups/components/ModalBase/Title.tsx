import React, {FC} from 'react';

import {ITitlePopup} from '../PopupBase';
import useStyles from '../../../styles/theme/hooks/useStyles';
import {Typography} from '../../../typography/components/Typography/Typography';

import stylesCreate from './stylesCreate';

/**
 * @deprecated use TextContent.tsx
 */

const Title: FC<ITitlePopup> = props => {
  const {title, titleStyles, titleFont} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <Typography
      style={[styles.text, titleStyles]}
      font={titleFont ? titleFont : 'SemiBold-Primary-L'}>
      {title}
    </Typography>
  );
};

export default Title;
