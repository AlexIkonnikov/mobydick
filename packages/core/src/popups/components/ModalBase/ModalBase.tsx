import {FC, PropsWithChildren} from 'react';
import {StyleProp, ViewStyle} from 'react-native';

import {PopupBase} from '../PopupBase';
import {IPopup} from '../../types';
import {returnTrue} from '../../functions';
import useStyles from '../../../styles/theme/hooks/useStyles';
import View from '../../../basic/components/View/View';

import CloseIcon from './CloseIcon';
import VerticalButtonsView from './VerticalButtonsView';
import HorizontalButtonsView from './HorizontalButtonsView';
import AlertContent from './AlertContent';
import VerticalButton from './VerticalButton';
import ImageView from './ImageView';
import stylesCreate from './stylesCreate';
import TextContent from './TextContent';

const ModalBase: FC<
  PropsWithChildren<
    Omit<IPopup, 'Content'> & {
      overlayStyle?: StyleProp<ViewStyle>;
      onClose: () => void;
      containerStyle?: StyleProp<ViewStyle>;
    }
  >
> & {
  CloseIcon: typeof CloseIcon;
  VerticalButtonsView: typeof VerticalButtonsView;
  HorizontalButtonsView: typeof HorizontalButtonsView;
  VerticalButton: typeof VerticalButton;
  AlertContent: typeof AlertContent;
  ImageView: typeof ImageView;
  TextContent: typeof TextContent;
} = props => {
  const {children, overlayStyle, onClose, containerStyle} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <PopupBase
      onClose={onClose}
      overlayStyle={[styles.overlayStyle, overlayStyle]}>
      <View
        style={[styles.container, containerStyle]}
        onStartShouldSetResponder={returnTrue}>
        {children}
      </View>
    </PopupBase>
  );
};

ModalBase.CloseIcon = CloseIcon;
ModalBase.VerticalButtonsView = VerticalButtonsView;
ModalBase.HorizontalButtonsView = HorizontalButtonsView;
ModalBase.AlertContent = AlertContent;
ModalBase.VerticalButton = VerticalButton;
ModalBase.ImageView = ImageView;
ModalBase.TextContent = TextContent;
export default ModalBase;
