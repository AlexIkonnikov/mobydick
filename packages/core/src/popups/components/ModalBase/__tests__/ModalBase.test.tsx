import React from 'react';
import {render} from '@testing-library/react-native';

import ModalBase from '../ModalBase';

describe('@lad-tech/mobydick-core/ModalBase', () => {
  afterEach(() => {
    jest.resetAllMocks();
    jest.clearAllMocks();
  });
  it('should renders correctly', () => {
    const {toJSON} = render(<ModalBase id={'id'} onClose={() => null} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
