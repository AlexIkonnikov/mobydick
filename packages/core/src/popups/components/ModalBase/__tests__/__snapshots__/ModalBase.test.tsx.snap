// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/ModalBase should renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      [
        {
          "justifyContent": "flex-end",
        },
        undefined,
      ],
    ]
  }
  testID="Pressable"
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 18,
          "justifyContent": "center",
          "marginVertical": 30,
          "paddingHorizontal": 30,
          "paddingVertical": 36,
          "width": 726,
        },
        undefined,
      ]
    }
  />
</View>
`;
